### DISCUSSION TITLE
Вопросы по редактированию веб страницы

### DISCUSSION THREAD

**Discussion by efremych** [Chimildic/goofy#109] [Помощь с алгоритмом] [answered]
1) У меня, в отличие от примера, добавляется в начале текста "Результат:".
2) Не сохраняется форматирование текста, а именно - теряются `'\n'`
Эти моменты можно победить?
![image](https://user-images.githubusercontent.com/76212904/126908541-06e2bd55-e7ac-4b23-b399-6fb497681260.png)

3) Это актуально? А то у меня чего-то не получилось так.
![image](https://user-images.githubusercontent.com/76212904/126908649-43ef6a67-1f65-4608-8380-913d33b83d54.png)

4) И самое главное. Как можно получить информацию по `id `трека? А-ля `Get a Track`. 
Ведь мы можем передавать `id `трека в форму, как потом подтянуть всю остальную информацию по треку?

Времени, к сожалению, сейчас немного, самостоятельно не удалось одолеть п.4 :(

**Comment by Chimildic** [comment, marked as answer]
1. В `launch.html` функции `onSuccess`, `onFailure`. Можно убрать это слово.
2. Особенность вывода в `html`, все в одну строчку. Могу поправить.
3. Да, актуально. Что не получается?
4. Для чего получать весь трек? Большая часть функций только `id` смотрит. Просто упаковать в объект `{ id: trackId }` или массив `[{ id: trackId }]`. Полный трек:
```js
let trackId = Helper.parseId(values.text.trackUrl);
let track = SpotifyRequest.getFullObjByIds('tracks', [trackId], 50)[0];
```

**Comment by efremych** [comment]
`В launch.html функции onSuccess, onFailure. Можно убрать это слово.`

Спасибо, блин, почему-то не нашел ctr+f, когда искал там....
Кстати, одна из причин, которая меня останавливает пользоваться обновленной версией AppScripts - что там нельзя искать по всему проекту (если у меня несколько скриптов). Это так или я что неправильно делаю?

`Особенность вывода в html, все в одну строчку. Могу поправить.`

Если это не супер трудозатратно, мне кажется, было бы полезно с точки зрения визуализации результатов...

`Да, актуально. Что не получается?`

Странно, у меня мелкий текст в итоге и без кнопки... Ну раз должно работать, потом вернусь, уже отказался от простого вызова без форм.

`let track = SpotifyRequest.getFullObjByIds('tracks', [trackId], 50)[0];`

Спасибо. Видимо, один шажок не дошел. Уже и функцию эту использовал и айди в массив засунул. Оставалось, похоже, только обратиться через [0]. эх. А может не один...

`Для чего получать весь трек?`

Сейчас хочу сделать чек играющего или введенного в текстовое поле трека на предмет наличия трека в моих плейлистах. Трек, введенный вручную, может не найтись в плейлистах. А раз уж я для себя уже решил что во всех случаях буду выводить название трека, то откуда мне его еще взять... Не отказываться же от прихоти :)

**Reply by Chimildic** [reply]
> нельзя искать по всему проекту

Видимо не умеет по всем файлам искать. Но в сравнении со старым вариантом, куда удобнее в других аспектах.

> мелкий текст в итоге и без кнопки

Сам текст является кнопкой в таком случаи. [Данный](https://github.com/Chimildic/goofy/discussions/9#discussioncomment-502624) вариант удобнее.

**Reply by efremych** [reply]
`Сам текст является кнопкой в таком случаи`

О, и вправду.

`Но в сравнении со старым вариантом, куда удобнее в других аспектах`

Так-то, наверное, да. Но старый вариант как-то привычнее, роднее уже)
Но я уже понял, что скоро насильно переведут - не получалось обновить права доступа. А в новой версии - получилось.
Автозамена _(если цветовую схему использовать стандартную - иначе не увидишь что пишешь)_ по проекту хваленая тоже по-моему не работает  - меняет по какой-то своей логике.
Но за подсказки по новой версии - спасибо, много полезного.

**Comment by Chimildic** [comment]
По поводу переноса строки. В файле `launch.html` добавить `#logger{white-space:pre-wrap}` в блок `style`.

![2021-07-26_18-08-16](https://user-images.githubusercontent.com/40138097/127012954-65989bde-003b-4cb7-bd16-3de22a847f35.png)

**Reply by efremych** [reply]
Спасибо! Позднее проверю.

**Reply by efremych** [reply]
Спасибо, работает
